src/components/plugin/glossary-popup.tsx

```diff
@@ -90,18 +91,13 @@ export default class GlossaryPopup extends React.Component<IProps, IState> {
 
   public render() {
     const { questionVisible } = this.state;
-    const { secondLanguage, onLanguageChange } = this.props;
+    const { otherLanguages, onLanguageChange } = this.props;
     return (
       <div className={css.glossaryPopup}>
-        {
-          secondLanguage && onLanguageChange &&
-          <Button
-            data-cy="langToggle"
-            className={css.langButton}
-            label={POEDITOR_LANG_NAME[secondLanguage].replace("_", " ")}
-            onClick={onLanguageChange}
-          />
-        }
+        <LanguageSelector
+          otherLanguages={otherLanguages}
+          onLanguageChange={onLanguageChange}
+        />
```

Comment:
This doesn't look conditional on the presence of otherLanguages.  If a glossary doesn't have other languages, or it doesn't allow student switching between languages then this LanguageSelector should not be shown.

Reply:
To clarify further, even if the glossary doesn't let students choose their own languages, it is possible the teacher will enable an alternative language for a specific student, so in that case the Language Selector would be shown.

Reply:
I see it is handled in the component itself.